### ISSUE TITLE
你好，我觉得你这个代码写错了，和你论文里的扫描方式并不是一致，论文里你写的是交错跨模态，但你这个代码里写的是先扫描完一行RGB，再扫描一行另一个模态，并不对

### ISSUE THREAD

**Issue by TomorrowJW** [xiaodonguo/CMSSM#2]

**Comment by xiaodonguo** [comment]
你可以查看fusion.py代码中的414-418行，torch.stack即将两个模态的各个像素交错。不需要看备注，有些备注是多余的。

**Comment by TomorrowJW** [comment]
你好，我看了，并使用GPT5分析，看了一下午，你的这个确实不对，假设序列R=[R1,R2,R3,R4]和T=[T1,T2,T3,T4]，你论文的扫描方式是[R1,T1,R2,T2,R3,T3,R4,T4]，但你代码里的是[R1,R2,R3,R4,T1,T2,T3,T4]，并不是交错，而是把每个模态展开一维，然后拼接，您自己再检查一遍？兄弟，不过你的工作还是很出色的，你的硕士团队周老师的论文我读了很多篇，很有收获，感谢你的及时回复！

**Comment by TomorrowJW** [comment]
你代码主要错在stack在最后一维，应该是必须在通道维之后、新增模态维（dim=2），谢谢，非常认可你的工作，但这不影响这是一篇高质量的Paper!

**Comment by xiaodonguo** [comment]
感谢您对我们工作的认可和关注。这篇工作是一年前完成的，所以我花了一段时间重新检查了我的代码，得到的结论是代码中的逻辑并没有错误。你可以用简单的方法进行检验，例如赋值一个(B,C,L)形状全为0的向量和一个(B,C,L)形状全为1的向量，在dim=-1的维度进行stack,检查输出的结果是否为(0,1,0,1,...,0,1)。若按你所说的在dim=1(通道维度之后)上stack反而会得到[000,111,000,111]的结果。我觉得可能是您对stack的操作的理解存在偏差，如果还有其他问题可以发送到我的邮箱3120245534@bit.edu.cn

**Comment by TomorrowJW** [comment]
不好意思，我刚自己试验了一下我的方法，和你方法都是对的，深感抱歉，感谢回复，我的先在第二个维度堆叠，然后又permute了一下，这个stack确实有点绕，能否加个微信，以后合作科研？非常感谢

import torch

B, C, H, W = 1, 1, 2, 2
L = H * W

# 构造 RGB 和 D 的 toy example
rgb = torch.tensor([[[[1, 2],
                      [3, 4]]]])   # shape (1,1,2,2)
d   = torch.tensor([[[[5, 6],
                      [7, 8]]]])   # shape (1,1,2,2)

print("rgb:", rgb.size())
print("d:", d.size())

# -------- 行展开 (H,W -> L) --------
rgb_hw = rgb.view(B, -1, L)  # (B,C,L)
d_hw   = d.view(B, -1, L)
print("rgb_hw:", rgb_hw)
print("d_hw:", d_hw)
print("-"*50)

# ================== 方法1: stack dim=2，然后 permute ==================
seq_dim2 = torch.stack([rgb_hw, d_hw], dim=2)   # (B,C,2,L)
print("[dim=2] stack shape:", seq_dim2.shape)
print(seq_dim2)

seq_dim2_perm = seq_dim2.permute(0,1,3,2).contiguous()  # (B,C,L,2)
seq_dim2_flat = seq_dim2_perm.reshape(B, -1, 2*L)
print("flatten after dim=2+permute (交错):", seq_dim2_flat)
print("交错序列 (dim=2):", seq_dim2_flat.view(-1).tolist())
print("-"*50)

# ================== 方法2: stack dim=-1 ==================
seq_last = torch.stack([rgb_hw, d_hw], dim=-1)   # (B,C,L,2)
print("[dim=-1] stack shape:", seq_last.shape)
print(seq_last)

seq_last_flat = seq_last.reshape(B, -1, 2*L)
print("flatten after dim=-1 (交错):", seq_last_flat)
print("交错序列 (dim=-1):", seq_last_flat.view(-1).tolist())
print("-"*50)

# ================== 总结 ==================
print("期望交错序列: [1, 5, 2, 6, 3, 7, 4, 8]")

**Comment by xiaodonguo** [comment]
您可以把微信发到我邮箱，我加您。
